src/model/user-course.model.ts

```diff
@@ -0,0 +1,36 @@
+import { Entity, Column, PrimaryGeneratedColumn, CreateDateColumn, UpdateDateColumn, DeleteDateColumn } from 'typeorm'
+
+import { UserCourseLevel } from 'devu-shared-modules'
+
+@Entity('user-course')
+export default class UserCourseModel {
+  @PrimaryGeneratedColumn()
+  id: number
+
+  @CreateDateColumn({ name: 'created_at' })
+  createdAt: Date
+
+  @UpdateDateColumn({ name: 'updated_at' })
+  updatedAt: Date
+
+  @DeleteDateColumn({ name: 'deleted_at' })
+  deletedAt?: Date
+
+  // Foreign key
+  @Column({ name: 'user_id' })
```

Comment:
Alrighty, I did some testing and I'm pretty sure this is what we want. This allows us to actually have the foreign key constraint on the column with the auto migration. This won't query the user table at all when querying user_course, but it will yell at you if you attempt to insert with a not there `userId`

```suggestion
  @Column({ name: 'user_id' })
  @JoinColumn({ name: 'user_id' })
  @ManyToOne(() => UserModel)
  userId: number
```

Obviously we'll probably want to add this to the course_id as well. But we'll have to wait for the courses model PR to go in first then update this.

Reply:
This is hot! All the docs use UserModel as the type so I thought that was required. I'm so glad it's as simple as making it a number!